src/pt/yugenmangas/src/eu/kanade/tachiyomi/extension/pt/yugenmangas/YugenMangasDto.kt

```diff
@@ -1,83 +1,112 @@
 package eu.kanade.tachiyomi.extension.pt.yugenmangas
 
+import eu.kanade.tachiyomi.extension.pt.yugenmangas.YugenMangas.Companion.DATE_FORMAT
 import eu.kanade.tachiyomi.source.model.SChapter
 import eu.kanade.tachiyomi.source.model.SManga
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.json.JsonNames
-import java.text.SimpleDateFormat
-import java.util.Locale
+import java.util.Calendar
 
 @Serializable
-class YugenMangaDto(
+class PageDto<T>(
+    @SerialName("current_page")
+    val page: Int,
+    @SerialName("total_pages")
+    val totalPages: Int,
+    val results: List<T>,
+) {
+    fun hasNext() = page < totalPages
+}
+
+@Serializable
+class MangaDto(
+    @JsonNames("series_code")
+    val code: String,
+    @JsonNames("path_cover")
+    val cover: String,
+    @JsonNames("title", "series_name")
```

Comment:
`@SerialName`?

Reply:
```@SerialName``` only works for fields with specific names. I'm using the same DTO in the popularManga(_code, cover, title_), latestUpdates(_series_code, path_cover, series_name_) and Search(_series_code, cover, name_), and ```@JsonNames``` supports multiple names